src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -1421,14 +1421,29 @@ private static (Vector128<byte>, Vector128<byte>, Vector128<byte>) FormatGuidVec
                     Vector128.Create(0x7060504FF030201, 0xF0E0D0C0B0A0908).AsByte());
 
                 // Vector "z" - we need to merge some elements of hexLow with hexHigh and add 4 dashes.
-                Vector128<byte> mid1 = Vector128.Shuffle(hexLow,
-                    Vector128.Create(0x0D0CFF0B0A0908FF, 0xFFFFFFFFFFFF0F0E).AsByte());
-                Vector128<byte> mid2 = Vector128.Shuffle(hexHigh,
-                    Vector128.Create(0xFFFFFFFFFFFFFFFF, 0xFF03020100FFFFFF).AsByte());
-                Vector128<byte> dashesMask = Vector128.Shuffle(Vector128.CreateScalarUnsafe((byte)'-'),
-                    Vector128.Create(0xFFFF00FFFFFFFF00, 0x00FFFFFFFF00FFFF).AsByte());
-
-                Vector128<byte> vecZ = (mid1 | mid2 | dashesMask);
+                Vector128<byte> vecZ;
+                Vector128<byte> dashesMask = Vector128.Create(0x00002D000000002D, 0x2D000000002D0000).AsByte();
+                if (AdvSimd.Arm64.IsSupported)
+                {
+                    // Arm64 allows shuffling values using a 32-byte wide look-up table consisting of two 128-bit registers.
+                    // Each byte in the second arg represents a value between 0 to 31 that acts as an index in the look-up table.
+                    // Now we can create a "z" vector by selecting 12 values starting from the 9th element (index 0x08) and
+                    // leaving gaps for dashes. Thus, the wider look-up table allows combining two shuffles, as used in the
+                    // generic else-case, into a single instruction on Arm64.
+                    // TODO: Check if the JIT can merge the consecutive table look-ups and avoid the Arm64 specific if-case.
```

Comment:
Did you intend to leave the todo? In libraries code we generally prefer to track work with issues.

Reply:
Hi @danmoseley, the todo is intentional here as it represents an investigation as well. I'm not sure whether if at all it could be done and how easy/hard it would be. Probably @tannergooding can suggest its feasibility.

Reply:
Removing the todo #88337 👍